### PR TITLE
{cae}[intel/2018a] Nek5000 v17.0

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by hajgato** [comment]
@boegel Can you tell me what is the problem?
```
======================================================================
FAIL: Test for parsing of easyconfig Nek5000-17.0-intel-2018a.eb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<string>", line 1, in innertest
  File "/home/travis/build/easybuilders/easybuild-easyconfigs/test/easyconfigs/easyconfigs.py", line 456, in template_easyconfig_test
    self.assertEqual(orig_val, dumped_val)
AssertionError: ['sed -i -e "s|^#FC=.*|FC=$FC|" bin/makenek', 'sed -i -e "s|^#CC=.*|CC=$CC|" bin/makenek', 'sed -i -e "s|^#FFLAGS=.*|FFLAGS=\\"$FFLAGS\\"|" bin/makenek', 'sed -i -e "s|^#CFLAGS=.*|CFLAGS=\\"$CFLAGS\\"|" bin/makenek'] != ['sed -i -e "s|^#FC=.*|FC=$FC|" bin/makenek', 'sed -i -e "s|^#CC=.*|CC=$CC|" bin/makenek', 'sed -i -e "s|^#FFLAGS=.*|FFLAGS="$FFLAGS"|" bin/makenek', 'sed -i -e "s|^#CFLAGS=.*|CFLAGS="$CFLAGS"|" bin/makenek']
```

File: easybuild/easyconfigs/n/Nek5000/Nek5000-17.0-intel-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    'mv %(name)s-%(version)s/{*,.[^.]*} . && rmdir %(name)s-%(version)s && cd tools && ./maketools all &&'
+    'cd ../3rd_party/gslib && env GSLIB_OPT="USREXIT=1 BLAS=2 PREFIX=gslib_ FPREFIX=fgslib_" ./install &&'
+    'cd ../cvode && ./install'
+)
+
+postinstallcmds = [
+    'sed -i -e "s|^#FC=.*|FC=$FC|" bin/makenek',
+    'sed -i -e "s|^#CC=.*|CC=$CC|" bin/makenek',
+    'sed -i -e "s|^#FFLAGS=.*|FFLAGS=\\"$FFLAGS\\"|" bin/makenek',
+    'sed -i -e "s|^#CFLAGS=.*|CFLAGS=\\"$CFLAGS\\"|" bin/makenek',
```
Comment (boegel) [inline easybuild/easyconfigs/n/Nek5000/Nek5000-17.0-intel-2018a.eb]: @hajgato The tests that verifies whether a dumped easyconfig file is fully equivalent with the original trips over this (the escaped inner double quotes), you may need to use a workaround like use `<doublequote>` first, which you then replace at the end with `"` in an additional command...

Comment (boegel) [reply easybuild/easyconfigs/n/Nek5000/Nek5000-17.0-intel-2018a.eb]: Why not include this in the patch file instead?

Comment (hajgato) [reply easybuild/easyconfigs/n/Nek5000/Nek5000-17.0-intel-2018a.eb]: @boegel I do not icluse this in a patch, becasue I need the value of `$FC`, `$CC`, `$FFLAG`, `$CFLAG` environmental variables.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
cermis - Linux debian 9.9, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/bab386bf54542eb453bc29c1a28fb128 for a full test report.

**Comment by vanzod** [comment]
@hajgato It looks like that it also needs `sundials`. Right now it is downloading it from `computation.llnl.gov` and when attempting to build it it fails with:
```
./install: line 20: cd: too many arguments
```
Ideally we would want to build it externally and add it as a dependency.

**Comment by hajgato** [comment]
@vanzod The problem is that `Nek5000` uses a bunch of external dependencies (`gslib 1.0.1`, `SUNDIALS 2.7.0`, `Hypre 2.11.2`, `netCDF 4.4.1.1`, `exodus`), some of them with special config. So originally I have decided to use the download and build locally approach. Not sure that other versions are working or not with `Nek5000`, so we might have some version clashes within the toolchain. So I would rather solve the problem by patching then adding `SUNDIALS` as a dep. The problem is that debian gives an error to `cd sundials-2.7.0/ sundials-2.7.0.tar.gz` meanwhile RHEL will change to the directory.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3402.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/7ee91bd77a58873b42bdcbd53cd9ab8d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2603.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/5529d821ee578b7b8b67137e4903e996 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**